src/commands/misc/man.rs

```diff
@@ -0,0 +1,137 @@
+use crate::{Context, Error};
+use poise::serenity_prelude::{
+    ComponentInteractionCollector, CreateActionRow, CreateButton, CreateInteractionResponse,
+    CreateInteractionResponseMessage,
+};
+use poise::CreateReply;
+use std::process::Command;
+use std::time::Duration;
+use tracing::{debug, warn};
+
+const MAN_VIEW_TIMEOUT: Duration = Duration::from_secs(10 * 60);
+const MSG_MAX_LEN: usize = 1950;
+
+fn remove_first_and_last_line(input: String) -> String {
+    let lines: Vec<&str> = input.lines().collect();
+    if lines.len() <= 2 {
+        return String::new();
+    }
+    lines[1..lines.len() - 1].join("\n")
+}
+
+#[poise::command(slash_command, prefix_command)]
+pub(crate) async fn man(ctx: Context<'_>, text: String) -> Result<(), Error> {
+    debug!("man {}", text);
+    let com = Command::new("man").arg(text).output()?;
+    let com_str = String::from_utf8_lossy(&com.stdout).parse()?;
+    let page = remove_first_and_last_line(com_str);
+
+    let ctx_id = ctx.id();
+    let prev_button_id = format!("{}prev", ctx.id());
+    let next_button_id = format!("{}next", ctx.id());
+
+    let mut idx: usize = 0;
+    let page_count = calculate_count(&page);
```

Comment:
Can you build the pages here already? then the paging becomes a simple array index.